### ISSUE TITLE
Shouldn't we make __toString() return type check less strict?

### ISSUE THREAD

**Issue by roberthusak** [peachpiecompiler/peachpie#386]
We force the return value of the ```__toString()``` method to be always string, because the PHP runtime requires it as well. E.g. for this code we report the error ```PHP4052: Method A::__toString() must return a string value```:
```php
class A {
  public function __toString() { return false; }
}
```
I wonder if it wouldn't be useful to (possibly by setting a certain compilation flag) to turn it into a warning and defer the error to the runtime. It seems that PHP programmers do something like ```return false``` from time to time if they are sure that the class is never used as a string. Or maybe they do it on purpose to discover such situations? I don't know. Anyway, doing it would enable us to compile more PHP code without any modification.

**Comment by jakubmisek** [comment]
returning anything else than `string` causes fatal error in PHP. 

But maybe we report it falsely sometimes? e.g.
```php
function __tostring() {
   raise_exception();
   return;
}

**Comment by roberthusak** [comment]
So far I've seen only justified reports of the error, but the false positives are possible, as the analysis will never be perfectly precise.

**Comment by jakubmisek** [comment]
introduced as a warning, old error `PHP4052` removed
